frontend/src/MuiForms/DataCardSelector.tsx

```diff
@@ -0,0 +1,155 @@
+      setSelectedDataCards(newValues)
+    },
+    [onChange],
+  )
+
+  const handleInputChange = useCallback((_event: SyntheticEvent<Element, Event>, value: string) => {
+    setDataCardListQuery(value)
+  }, [])
+
+  const debounceOnInputChange = debounce((event: SyntheticEvent<Element, Event>, value: string) => {
+    handleInputChange(event, value)
+  }, 500)
+
+  if (isDataCardsError) {
+    if (isDataCardsError.status !== 413) {
+      return <MessageAlert message={isDataCardsError.info.message} severity='error' />
+    }
+  }
+
+  return (
+    <>
+      {isDataCardsLoading && <Loading />}
+      {formContext && formContext.editMode && (
+        <Autocomplete<ModelSearchResult, true, true>
+          multiple
+          data-test='dataCardSelector'
+          loading={dataCardListQuery.length > 3 && isDataCardsLoading}
+          open={open}
+          size='small'
+          onOpen={() => {
+            setOpen(true)
+          }}
+          onClose={() => {
+            setOpen(false)
+          }}
+          disableClearable
+          isOptionEqualToValue={(option, value) => option.id === value.id}
+          getOptionLabel={(option) => option.name}
+          value={selectedDataCards || []}
+          filterOptions={(x) => x}
+          onChange={handleModelChange}
+          noOptionsText={dataCardListQuery.length < 3 ? 'Please enter at least three characters' : 'No options'}
+          onInputChange={debounceOnInputChange}
+          options={dataCards || []}
+          renderTags={(value, getTagProps) =>
+            value.map((option, index) => (
+              <Box key={option.name} sx={{ maxWidth: '200px' }}>
+                <Chip {...getTagProps({ index })} sx={{ textOverflow: 'ellipsis' }} label={option.name} />
+              </Box>
+            ))
+          }
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              placeholder='Data card name'
+              error={rawErrors && rawErrors.length > 0}
+              label={label + (required ? ' *' : '')}
+              onKeyDown={(event: KeyboardEvent) => {
+                if (event.key === 'Backspace') {
+                  event.stopPropagation()
+                }
+              }}
+            />
+          )}
+        />
+      )}
+      {formContext && !formContext.editMode && (
+        <>
+          <Typography fontWeight='bold'>
+            {label}
+            {required && <span style={{ color: theme.palette.error.main }}>{' *'}</span>}
+          </Typography>
+          {currentValue.length === 0 && (
+            <Typography
+              sx={{
+                fontStyle: 'italic',
+                color: theme.palette.customTextInput.main,
+              }}
+            >
+              Unanswered
+            </Typography>
+          )}
+          <Box sx={{ overflowX: 'auto', p: 1 }}>
+            <Stack spacing={1} direction='row'>
+              {currentValue.map((currentDataCard) => (
```

Comment:
`currentDataCard` is the data card ID, but this isn't clear. Consider renaming to `currentDataCardId`